pkg/proxystorage/proxy.go

```diff
@@ -57,13 +57,17 @@ func (p *proxyStorageState) Cancel(n *proxyStorageState) {
 }
 
 // NewProxyStorage creates a new ProxyStorage
-func NewProxyStorage() (*ProxyStorage, error) {
-	return &ProxyStorage{}, nil
+func NewProxyStorage(Memory int) (*ProxyStorage, error) {
+	ps := ProxyStorage{
+		MaxMemory: Memory,
```

Comment:
I don't like this max memory flag being passed down all these layers :/ it existing on the proxyStorage layer implies that it'll be enforced here but we just pass it down. Maybe this would be better suited in a configuration instead of a flag?